client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -76,6 +79,16 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
     JavaUtils.newConcurrentHashMap[Int, ConcurrentHashMap[Int, PartitionLocation]]()
   private val userIdentifier: UserIdentifier = IdentityProvider.instantiate(conf).provide()
 
+  private val rpcCacheSize = conf.clientRpcCacheSize
+  private val rpcCacheConcurrencyLevel = conf.clientRpcCacheConcurrencyLevel
+  private val rpcCacheExpireTime = conf.clientRpcCacheExpireTime
+
+  private val getRegisterShuffleResponseRpcCache: Cache[Int, ByteBuffer] = CacheBuilder.newBuilder()
+    .concurrencyLevel(rpcCacheConcurrencyLevel)
+    .expireAfterAccess(rpcCacheExpireTime, TimeUnit.MILLISECONDS)
+    .maximumSize(rpcCacheSize)
+    .build().asInstanceOf[Cache[Int, ByteBuffer]]
```

Comment:
Shouldn't we not invalidate cache entry for mutations to `workerSnapshots`

Reply:
@mridulm ```flatMap(_.getAllPrimaryLocationsWithMinEpoch())```  the minimum value of epoch will be taken here, so changes to workersnapshot will not affect the cache.